python/cudf/cudf/core/column/column.py

```diff
@@ -15,32 +15,186 @@
+        """
 
+        super().__init__(
+            data,
+            size=size,
+            dtype=dtype,
+            mask=mask,
+            offset=offset,
+            children=children,
+            name=name,
+        )
+        self._data = data
+        self.size = size
+        self.dtype = dtype
+        self._mask = mask
+        self.offset = offset
+        self.children = children
+        self.name = name
+
+    def __reduce__(self):
+        return (
+            build_column,
+            (
+                self.data,
+                self.dtype,
+                self.mask,
+                self.offset,
+                self.children,
+                self.name,
+            ),
+        )
 
-class Column(object):
-    """An immutable structure for storing data and mask for a column.
-    This should be considered as the physical layer that provides
-    container operations on the data and mask.
+    @property
+    def data(self):
+        return self._data
 
-    These operations work on each data element as plain-old-data.
-    Any logical operations are implemented in subclasses of *TypedColumnBase*.
+    @data.setter
+    def data(self, value):
+        self._data = value
 
-    Attributes
-    ----------
-    _data : Buffer
-        The data buffer
-    _mask : Buffer
-        The validity mask
-    _null_count : int
-        Number of null values in the mask.
-
-    These attributes are exported in the properties (e.g. *data*, *mask*,
-    *null_count*).
-    """
+    @property
+    def mask(self):
+        return self._mask
+
+    @mask.setter
+    def mask(self, value):
+        self._nvstrings = None
+        self._mask = value
+
+    def _data_view(self):
+        """
+        View the data as a device array or nvstrings object
+        """
+        if self.dtype == "object":
+            return self.nvstrings
+
+        if is_categorical_dtype(self.dtype):
+            return self.codes._data_view()
+        else:
+            dtype = self.dtype
+
+        result = rmm.device_array_from_ptr(
+            ptr=self.data.ptr, nelem=len(self), dtype=dtype
+        )
+        result.gpu_data._obj = self
+        return result
+
+    def _mask_view(self):
+        """
+        View the mask as a device array
+        """
+        result = rmm.device_array_from_ptr(
+            ptr=self.mask.ptr,
+            nelem=calc_chunk_size(len(self), mask_bitsize),
+            dtype=np.int8,
+        )
+        result.gpu_data._obj = self
+        return result
+
+    def __len__(self):
+        return self.size
+
+    def to_pandas(self):
+        arr = self._data_view()
+        sr = pd.Series(arr.copy_to_host())
+
+        if self.mask is not None:
+            mask_bits = self._mask_view().copy_to_host()
+            mask_bytes = (
+                cudautils.expand_mask_bits(len(self), mask_bits)
+                .copy_to_host()
+                .astype(bool)
+            )
+            sr[~mask_bytes] = None
```

Comment:
This will change the dtype to either `object` or `float`, I don't think that's good expected behavior.